components/bpmn/org.wso2.carbon.bpmn.uuf.ui.stub/src/main/java/org.wso2.carbon.bpmn.uuf.ui/service/BPMNExplorerServiceImpl.java

```diff
@@ -0,0 +1,477 @@
+     */
+    private JSONObject createProcessInstance(String processDefId, String username, String password) throws JSONException, IOException {
+        HttpPost httpPost = null;
+        String line = "";
+        StringEntity stringEntity;
+        JSONObject rsltObj;
+        JSONObject rtrnObj = new JSONObject();
+
+        try (CloseableHttpResponse response = null;
+             BufferedReader rd = null) {
+            httpPost = new HttpPost(
+                    baseUrl + "/bpmn/runtime/process-instances");
+            httpPost.addHeader(BPMNExplorerServiceConstants.CONTENT_TYPE, BPMNExplorerServiceConstants.JSON_CONTENT_TYPE);
+            String authString = username + ":" + password;
+            byte[] authEncBytes = Base64.encodeBase64(authString.getBytes());
+            httpPost.addHeader(BPMNExplorerServiceConstants.AUTHORIZATION, BPMNExplorerServiceConstants.BASIC_AUTHORIZATION + new String(authEncBytes));
+
+            JSONObject bodyObj = new JSONObject();
+            bodyObj.put("processDefinitionId", processDefId);
+            stringEntity = new StringEntity(bodyObj.toString());
+            httpPost.setEntity(stringEntity);
+            response = client.execute(httpPost);
+            rd = new BufferedReader(
+                    new InputStreamReader(response.getEntity().getContent(), Charset.defaultCharset()));
+            StringBuffer result = new StringBuffer();
+            while ((line = rd.readLine()) != null) {
+                result.append(line);
+            }
+
+            rsltObj = new JSONObject(result.toString());
+            if (rsltObj.has("id")) {
+                JSONObject processInstanceObj = new JSONObject();
+                processInstanceObj.put("process-def-id", rsltObj.get("processDefinitionId"));
+                processInstanceObj.put("process-instance-id", rsltObj.get("id"));
+                rtrnObj.put("processInstanceData", processInstanceObj);
+            } else if (rsltObj.has("errorMessage")) {
+                rtrnObj.put("error-message", rsltObj.get("errorMessage"));
+            }
+        } catch (ClientProtocolException | IOException e) {
+            log.error("Exception : ", e);
+        } finally {
+            if (httpPost != null) {
+                httpPost.releaseConnection();
+            }
+        }
+        return rtrnObj;
+    }
+
+    /**
+     * Create process instance using form data
+     *
+     * @param frmData
+     * @param host
+     * @param port
+     */
+    @Override
+    public JSONObject createProcessInstanceWithData(String frmData, String host, int port, String username, String password)
+            throws JSONException, IOException {
+        HttpPost httpPost = null;
+        String line;
+        StringEntity stringEntity;
+        JSONObject rsltObj;
+        JSONObject rtrnObj = new JSONObject();
+
+        //Decode the query string for form data and process definition id
+        String decodedStr = URLDecoder.decode(frmData, "UTF-8");
```

Comment:
"UTF-8" has been used in many places. Those types of strings can be declared as constants and use where necessary. This will make the code maintainability easy.